mlflow/tensorflow/__init__.py

```diff
@@ -887,6 +968,110 @@ def _log_early_stop_callback_metrics(callback, history, metrics_logger):
         if metric_key is not None:
             metrics_logger.record_metrics(restored_metrics, stopped_epoch + 1)
 
+    def _log_keras_model_with_warning(history, args):
+        try:
+            if log_models:
+                import types
+
+                def _get_input_data_slice():
+                    input_training_data_original = args[0]
+                    input_example_slice = None
+
+                    def _extract_from_sequence_or_generator():
+                        input_training_data = input_training_data_original
+                        if isinstance(input_training_data_original, types.GeneratorType):
+                            input_training_data = next(input_training_data_original)[0]
+                        elif isinstance(
+                            input_training_data_original, tensorflow.keras.utils.Sequence
+                        ):
+                            input_training_data = input_training_data_original[:][0]
+                        return input_training_data
+
+                    input_training_data = _extract_from_sequence_or_generator()
+                    if isinstance(input_training_data, np.ndarray):
+                        input_example_slice = input_training_data[:INPUT_EXAMPLE_SAMPLE_ROWS]
+                    elif isinstance(input_training_data, tensorflow.data.Dataset):
+                        steps = 1
+                        if history.params is not None and "steps" in history.params:
+                            steps = history.params["steps"]
```

Comment:
Nit: `steps = (history.params or {}).get("steps", 1)` should work as a simplification